### DISCUSSION TITLE
Upgrading to Serverpod 2.x.x result in serverpod Error in server

### DISCUSSION THREAD

**Discussion by adiltirur** [serverpod/serverpod#2829] [Q&A] [answered]
Hello,

I am trying to upgrade from Serverpod 1.2.6 to 2.1.3, but when I upgrade and push to staging, the server breaks, and all calls result in a 502 bad gateway. Is there an automated script that I can run to upgrade everything correctly?

Right now below are the steps I did.

1. Upgraded all serverpod-related package in pubspec
2. Upgraded deployment-aws.yml to use the new dart version
3. Upgraded init-script.sh to use the new dart version
4. Run serverpod generate to update all generated files
5. Created new migration.
6. Push to the staging branch

As soon as I push the server will throw an error, Anything else I should be doing? I would want to upgrade from my current version, any help is highly appreciated.

Thank you

**Comment by Isakdl** [comment]
502 bad gateway sounds like the server is unreachable from the load balancer. Can you check the logs and see if there is any error printed?

**Reply by adiltirur** [reply]
When i run 

`systemctl status serverpod.service`

I get the following

```
 serverpod.service - Serverpod server
   Loaded: loaded (/usr/lib/systemd/system/serverpod.service; disabled; vendor preset: disabled)
   Active: failed (Result: start-limit) since Tue 2024-10-08 13:15:27 UTC; 8s ago
  Process: 16786 ExecStart=/home/ec2-user/serverpod/active/lc_core_server/deploy/aws/scripts/run_serverpod (code=exited, status=253)
 Main PID: 16786 (code=exited, status=253)

Oct 08 13:15:26 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: serverpod.service: main process exited, code=exited, status=253/n/a
Oct 08 13:15:26 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: Unit serverpod.service entered failed state.
Oct 08 13:15:26 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: serverpod.service failed.
Oct 08 13:15:27 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: serverpod.service holdoff time over, scheduling restart.
Oct 08 13:15:27 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: Stopped Serverpod server.
Oct 08 13:15:27 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: start request repeated too quickly for serverpod.service
Oct 08 13:15:27 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: Failed to start Serverpod server.
Oct 08 13:15:27 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: Unit serverpod.service entered failed state.
Oct 08 13:15:27 ip-10-0-6-121.eu-central-1.compute.internal systemd[1]: serverpod.service failed.
```

**Reply by Isakdl** [reply]
Right, so what we can see here is that the server is crashing, hence why it cannot be reached. Can you get it to run locally? Would need more logs / details from the running process to know exactly what went wrong. 

But since you have played around with the dart version that sounds like a reasonable potential cause. E.g. the used dart version is not compatible with the built server.

**Reply by adiltirur** [reply]
I can successfully run locally, no issues at all, but as soon as I deploy to was then the server crashes.

Also then what is the way to make the server compatible with the dart version, because a dart upgrade is necessary when I am coming from 1.x.x to 2.x.x

Can you let me know how I can get you more logs also please, thank you.

**Reply by mevech** [reply]
this is most likely due to the version of dart installed on the machine is not working for the new version. what you need to do is to install a new version of dart.

try following [this](https://github.com/mevech/serverpod_docs/blob/main/docs/07-deployments/04-deploying-to-aws.md#configure-dart-version) then restart your service to see if it works out?  :)

**Reply by adiltirur** [reply]
I did that every single step and get the same error, is there someway of seeing a more verbose log to see what exactly went wrong?

**Reply by mevech** [reply]
have you try to inspect `serverpod.err` and `serverpod.log`?

**Reply by adiltirur** [reply]
serverpod.err only have this

Can't load Kernel binary: Invalid kernel binary format version.


and serverpod.log is empty

**Reply by adiltirur** [reply, marked as answer]
Okay I resolved the issue, just in case someone from future visit here


The problem was caused by having multiple versions of Dart installed on the server. All builds and compilations were done using the new Dart version (3.5.1), but I still had the previous version installed. As a result, when the server restarted and executed the run_serverpod script, it was using the older Dart version, which led to the issue. Once I updated the Dart version that the run_serverpod script used, the errors disappeared.

Thank you, @Isakdl and @mevech , for guiding me in the right direction!
